File: 4ºAno/2ºSemestre/Desenvolvimento Mobile 2/Atividade04/Front/front/lib/models/store.dart
class Store {
  final String name;
  final String type;
  final String cep;
  final String logradouro;
  final String bairro;
  final String localidade;
  final String uf;

  Store({
    required this.name,
    required this.type,
    required this.cep,
    required this.logradouro,
    required this.bairro,
    required this.localidade,
    required this.uf,
  });

  factory Store.fromJson(Map<String, dynamic> json) {
    return Store(
      name: json['name'],
      type: json['type'],
      cep: json['cep'],
      logradouro: json['logradouro'],
      bairro: json['bairro'],
      localidade: json['localidade'],
      uf: json['uf'],
    );
  }
}

File: 4ºAno/2ºSemestre/Desenvolvimento Mobile 2/Atividade04/Api/WebApplication1/Models/Store.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;

namespace WebApplication1.Models;

public class Store
{
    public Store(string name, string type, string cep, string logradouro, string bairro, string localidade, string uf)
    {
        Name = name;
        Type = type;
        Cep = cep;
        Logradouro = logradouro;
        Bairro = bairro;
        Localidade = localidade;
        Uf = uf;
    }

    [Key]
    [JsonIgnore]
    public long Id { get; set; }

    public string Name { get; set; }

    public string Type { get; set; }

    public string Cep { get; set; }

    public string Logradouro { get; set; }

    public string Bairro { get; set; }

    public string Localidade { get; set; }

    public string Uf { get; set; }
}